### DISCUSSION TITLE
pinout for esp32

### DISCUSSION THREAD

**Discussion by surajchauhan6631** [forkineye/ESPixelStick#498] [General]
I am Very Much confused about which pinout of esp32 should be used to control pixel led and also to generate dmx signal. and through same pin can i control both ws2811 and dmx lights.  and also wanna know which pin should of esp32 should be used to attach a sd card with esp32. please help and guide me.

**Comment by MartinMueller2003** [comment]
Go to the admin page. Near the bottom is a checkbox marked "Advanced Mode". Set it to "Checked". Click on the device configuration page on the menu bar. You will now see the GPIO pins being used for the various outputs that have been selected. 

You cannot send DMX and WS2811 out the same port. They need to be on separate ports.

**Reply by surajchauhan6631** [reply]
Thanks Martin for your response. But  I am not able to find the advanced mode check box in admin page. And actually I didn't have too much knowledge to GitHub . So please help me so that I know which gpio pin is used for the various outputs

**Reply by surajchauhan6631** [reply]
Or are you talking about admin web page in which we do configuration of espixelstick

**Comment by MartinMueller2003** [comment]
At the bottom of the picture you can see the checkbox:

![image](https://user-images.githubusercontent.com/8608503/161433373-921e7f58-433f-43f4-8984-abb2090a28d0.png)

**Comment by MartinMueller2003** [comment]
And this is what the config screen will look like. Note the GPIO / pin settings being displayed:

![image](https://user-images.githubusercontent.com/8608503/161433446-1693b76a-b746-4f8d-8d5c-1f7def0b8ae6.png)

**Comment by Mat-Moo** [comment]
DMX cable has 3 pins, how does that connect to esp, advanced only shows 1 gpio output, so where do the other 2 cables connect too, and which of the 3 goes to the defined pin?

**Reply by MartinMueller2003** [reply]
None of the three DMX cable wires goes to the ESP. The ESP GPIO Output must go to a ttl-to-RS-485 adapter, along with 5v and V- from the ESP. The adapter will have a D+, D- and GND output Those three will go to the DMX cable.

FYI: The EspixelStick V3 HW does this for you. For that platform you would use the D+, D-, V- connections on the big green plug.

**Reply by Mat-Moo** [reply]
Ahhhhh thanks @MartinMueller2003 I have a v2 and was struggling to get it to work, so thought I'd try a d1 mini - Thanks for the info, I've a couple of ttl adapters I'll have a look

**Reply by Mat-Moo** [reply]
Got it working and connected to xlights, although seems a bit laggy (running a flood light) using Ddp and a beamz bac404 can as a test. I’ll sort a wiring diagram out/pic, but is laggy expected?

**Comment by MartinMueller2003** [comment]
WiFi can be laggy. Many people use a tuned WiFi network or turn off OTA Data and use fseq files stored on a local SD card.

**Reply by Mat-Moo** [reply]
Nope, run whole show with 5k leds on this unit, and for testing it’s the only thing currently connected… I’m thinking either Dmx terminator, or my wire hack to the Dmx is not great
